libraries/rush-lib/src/api/Subspace.ts

```diff
@@ -266,6 +266,27 @@ export class Subspace {
     return this._commonVersionsConfiguration;
   }
 
+  /**
+   * Gets the ensureConsistentVersions property from the common-versions.json config file,
+   * or from the rush.json file if it isn't defined in common-versions.json
+   * @beta
+   */
+  public get shouldEnsureConsistentVersions(): boolean {
+    if (
+      this.getCommonVersions().ensureConsistentVersions !== undefined &&
+      this._rushConfiguration.ensureConsistentVersions
```

Comment:
If **rush.json** has `"ensureConsistentVersions": false` and a subspace has `"ensureConsistentVersions": true`, this logic will not report an error.  Also this error is only reported at the time when the `ensureConsistentVersions` property getter is read for a subspace, which could be in the middle of some other operation.

Suppose we imposed stricter rules:

- If `subspacesEnabled=true`, then `ensureConsistentVersions` is forbidden in **rush.json**
- If `subspacesEnabled=false`, and `ensureConsistentVersions` is true in **rush.json**, then `ensureConsistentVersions`  is forbidden in `this.defaultSubspace.getCommonVersions()` (which gets loaded from the `common/config/rush` folder)

With this approach:
- The validation can be performed in the `RushConfiguration` constructor, which is always checked and will report the error during startup
- We don't even need to load **common-versions.json** unless the deprecated setting is found in **rush.json**
-  The  `this._rushConfiguration.ensureConsistentVersions` API continues still returns `false` (instead of `undefined`) when the setting is absent from **rush.json**, but the validation isn't broken because the constructor can inspect `IRushConfigurationJson` directly

Reply:
Changed